paymentsheet/src/main/java/com/stripe/android/paymentsheet/paymentdatacollection/ach/USBankAccountFormViewModel.kt

```diff
@@ -397,59 +404,72 @@ internal class USBankAccountFormViewModel @Inject internal constructor(
-                        name.value,
-                        email.value
-                    )
-                )
-            }
+            collectBankAccountForIntent(clientSecret)
         } else {
-            // Decoupled Flow
-            args.stripeIntentId?.let { elementsSessionId ->
-                if (args.isPaymentFlow) {
-                    collectBankAccountLauncher?.presentWithDeferredPayment(
-                        publishableKey = lazyPaymentConfig.get().publishableKey,
-                        stripeAccountId = lazyPaymentConfig.get().stripeAccountId,
-                        configuration = CollectBankAccountConfiguration.USBankAccount(
-                            name.value,
-                            email.value
-                        ),
-                        elementsSessionId = elementsSessionId,
-                        customerId = null,
-                        onBehalfOf = args.onBehalfOf,
-                        amount = args.formArgs.amount?.value?.toInt(),
-                        currency = args.formArgs.amount?.currencyCode
-                    )
-                } else {
-                    collectBankAccountLauncher?.presentWithDeferredSetup(
-                        publishableKey = lazyPaymentConfig.get().publishableKey,
-                        stripeAccountId = lazyPaymentConfig.get().stripeAccountId,
-                        configuration = CollectBankAccountConfiguration.USBankAccount(
-                            name.value,
-                            email.value
-                        ),
-                        elementsSessionId = elementsSessionId,
-                        customerId = null,
-                        onBehalfOf = args.onBehalfOf,
-                    )
-                }
-            }
+            collectBankAccountForDeferredIntent()
+        }
+    }
+
+    private fun collectBankAccountForIntent(clientSecret: String) {
+        val configuration = if (args.instantDebits) {
+            CollectBankAccountConfiguration.InstantDebits(
+                email = email.value,
+            )
+        } else {
+            CollectBankAccountConfiguration.USBankAccount(
+                name = name.value,
+                email = email.value,
+            )
+        }
+
+        if (args.isPaymentFlow) {
+            collectBankAccountLauncher?.presentWithPaymentIntent(
+                publishableKey = lazyPaymentConfig.get().publishableKey,
+                stripeAccountId = lazyPaymentConfig.get().stripeAccountId,
+                clientSecret = clientSecret,
+                configuration = configuration,
+            )
+        } else {
+            collectBankAccountLauncher?.presentWithSetupIntent(
+                publishableKey = lazyPaymentConfig.get().publishableKey,
+                stripeAccountId = lazyPaymentConfig.get().stripeAccountId,
+                clientSecret = clientSecret,
+                configuration = configuration,
+            )
+        }
```

Comment:
I know this was here already, but maybe we should gracefully fail if `collectBankAccountLauncher` is null?

Reply:
I’d prefer to leave as-is for now, and flag it with the Payments team. We currently fail by just doing nothing, so the user is still able to select another payment method without being thrown out of the flow.